### PR TITLE
Controllers1

### PR DESCRIPTION
**Pull request by kennyegun24**
## In this PR, I did the following...
- Created methods to allow users to create posts, write comments on posts and like posts
- Created methods to delete posts, comments and unlike posts
- Create methods to let current user to add work experiences

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Api::V1::PostsController < ApplicationController
+    skip_before_action :authenticate_request, only: %i[index show]
+    def create
+        @post = current_user.posts.create(post_params)
+
+        if @post.save
+            render json: {status: 'Successful', message: 'Post created'}, status: 201
+        else
+            render json: {status: 'Failed', message: 'Failed to create post'}, status: 422
```
Comment (Kamba56) [inline app/controllers/api/v1/posts_controller.rb:9]: - I think you should also display full error message here.

File: app/controllers/api/v1/work_experience_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class Api::V1::WorkExperienceController < ApplicationController
+    def create
+        @WorkExp = current_user.work_experiences.new(work_params)
+
+        if @WorkExp.save
+            render json: {status: 'Successful', message: 'Work Experience Added'}, status: 201
+        else
+            render json: {status: 'Failure', message: 'Something went wrong'}, status: 422
```
Comment (Kamba56) [inline app/controllers/api/v1/work_experience_controller.rb:8]: - Should also display full error messages.

File: app/controllers/api/v1/work_experience_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        end
+    end
+
+    def index
+        @user_work = current_user.work_experiences
```
Comment (Kamba56) [inline app/controllers/api/v1/work_experience_controller.rb:13]: - I think work experiences should be rendered in ascending or descending start_date order. I think handling that here will be more efficient than handling in the frontend.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
HI @kennyegun24, Great job so far. THere are some suggestions I think you should look into.

File: app/controllers/api/v1/post_comments_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            end
+        else
+            render json: { status: 'Failure', message: 'Comment not found' }, status: 404
+        end
+    end
+
+    def index
+        @comment = PostComment.where(post_id: params[:id])
+        render json: { status: 'Success', message: 'Successful', data: @comment }, status: 200
+    end
```
Comment (Kamba56) [inline app/controllers/api/v1/post_comments_controller.rb:31]: - There are no params passed to the index route. I understand why you used the show action in the first place. You can change back to the show action.

**Review by Kamba56 — APPROVED** [review]
@kennyegun24, Great job on implementing API endpoints.
Keep it up Agba Developer.

File: app/controllers/api/v1/post_comments_controller.rb (hunk 4 above)
Comment (kennyegun24) [reply app/controllers/api/v1/post_comments_controller.rb:31]: Shebi you see! No dey follow me argue next time! wicked reviewer

**Pull request merged**